### PR TITLE
style: remove pylint

### PR DESCRIPTION
**Pull request by lengau**
Pylint is pretty much entirely replaced with ruff at this point.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: snapcraft/commands/status.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def _get_channel_line(
     ]
 
 
-def _get_channel_lines_for_channel(  # noqa: C901  # pylint: disable=too-many-locals
+def _get_channel_lines_for_channel(  # noqa: C901
```
Comment (mr-cal) [inline snapcraft/commands/status.py:186]:
```suggestion
def _get_channel_lines_for_channel(  # noqa: C901 (complex-structure)
```

File: snapcraft/commands/status.py [hunk 2]
```diff
@@ -305,7 +305,7 @@ def _has_channels_for_architecture(
     return found_architecture
 
 
-def get_tabulated_channel_map(  # pylint: disable=too-many-branches, too-many-locals  # noqa: C901
+def get_tabulated_channel_map(  # noqa: C901
```
Comment (mr-cal) [inline snapcraft/commands/status.py:308]:
```suggestion
def get_tabulated_channel_map(  # noqa: C901 (complex-structure)
```

File: snapcraft/elf/_elf_file.py [hunk 3]
```diff
@@ -232,9 +232,7 @@ def is_elf(cls, path: Path) -> bool:
         with path.open("rb") as bin_file:
             return bin_file.read(4) == b"\x7fELF"
 
-    # pylint: disable=too-many-branches
-
-    def _extract_attributes(self) -> None:  # noqa: C901
+    def _extract_attributes(self) -> None:  # noqa: C901,PLR0912
```
Comment (mr-cal) [inline snapcraft/elf/_elf_file.py:235]:
```suggestion
    def _extract_attributes(self) -> None:  # noqa: C901,PLR0912 (complex-structure, too-many-branches)
```

File: snapcraft/parts/lifecycle.py [hunk 4]
```diff
@@ -114,7 +114,7 @@ def run(command_name: str, parsed_args: "argparse.Namespace") -> None:
         )
 
 
-def _run_command(  # noqa PLR0913 # pylint: disable=too-many-branches, too-many-statements
+def _run_command(  # noqa PLR0913
```
Comment (mr-cal) [inline snapcraft/parts/lifecycle.py:117]:
```suggestion
def _run_command(  # noqa PLR0913 (too-many-arguments)
```

File: snapcraft/parts/grammar.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ def process_parts(
     """
 
     def self_check(value: Any) -> bool:
-        return value == value  # pylint: disable=comparison-with-itself  # noqa PLR0124
+        return value == value  # noqa: PLR0124
```
Comment (mr-cal) [inline snapcraft/parts/grammar.py:69]:
```suggestion
        return value == value  # noqa: PLR0124 (comparison-with-itself)
```

File: tests/unit/parts/test_grammar.py [hunk 6]
```diff
@@ -26,7 +26,7 @@
 _PROCESSOR = GrammarProcessor(
     arch="amd64",
     target_arch="amd64",
-    checker=lambda x: x == x,  # pylint: disable=comparison-with-itself  # noqa PLR0124
+    checker=lambda x: x == x,  # noqa: PLR0124
```
Comment (mr-cal) [inline tests/unit/parts/test_grammar.py:29]:
```suggestion
    checker=lambda x: x == x,  # noqa: PLR0124 (comparison-with-itself)
```

**Review by mr-cal — APPROVED** [review]
Thanks!  I added a few rule names to the `noqa`s as suggestions.

**Review by cmatsuoka — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]

File: snapcraft/commands/unimplemented.py
[the remark this answers is not in the document]
Comment (mr-cal) [reply snapcraft/commands/unimplemented.py:1]: and that should fix the linter failures

[the remark this answers is not in the document]
Comment (lengau) [reply snapcraft/commands/unimplemented.py:1]: Yeah, thanks - I did the wrong thing with that conflict. Fixed!

**Pull request merged**